kore/src/Kore/Internal/SideCondition.hs

```diff
@@ -94,94 +97,115 @@ instance TopBottom (SideCondition variable) where
 instance InternalVariable variable
     => HasFreeVariables (SideCondition variable) variable
   where
-    freeVariables sideCondition@(SideCondition _) =
-        freeVariables assumedTrue
-      where
-        SideCondition {assumedTrue} = sideCondition
+    freeVariables = freeVariables . toPredicate
 
 instance InternalVariable variable => Unparse (SideCondition variable) where
-    unparse sideCondition@(SideCondition _) =
-        unparse assumedTrue
-      where
-        SideCondition {assumedTrue} = sideCondition
-
-    unparse2 sideCondition@(SideCondition _) =
-        unparse2 assumedTrue
-      where
-        SideCondition {assumedTrue} = sideCondition
+    unparse = unparse . toPredicate
+    unparse2 = unparse2 . toPredicate
 
-instance From (Condition variable) (SideCondition variable)
+instance From (SideCondition variable) (MultiAnd (Predicate variable))
   where
-      from = SideCondition
+    from condition@(SideCondition _) = assumedTrue condition
+    {-# INLINE from #-}
```

Comment:
I'm not entirely sure about these `INLINE` pragmas for the `From` instances here. I added them back, but I don't know if that's what we want.